Cannot enable SSL with OpenSSL 1.1.0c

Installed version: OpenSSL 1.1.0c-2 (Debian SID), after downgrade to 1.0.2j-4, everything runs smoothly.

```
$ ./configure --with-openssl
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... no
checking for mawk... mawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking for style of include used by make... GNU
checking dependency style of gcc... gcc3
checking for pthread_create in -lpthread... yes
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking whether to build with rdynamic for GNU ld... yes
checking for SSL_library_init in -lssl... no
configure: error: ssl library missing
```

Related part of config.log:
```
configure:4219: checking for SSL_library_init in -lssl
configure:4244: gcc -o conftest  -pthread   conftest.c -lssl  -lpthread  >&5
/tmp/cctFimQv.o: In function `main':
conftest.c:(.text+0xa): undefined reference to `SSL_library_init'
collect2: error: ld returned 1 exit status
configure:4244: $? = 1
```

--- a/configure.ac
+++ b/configure.ac
@@ -49,8 +49,8 @@ AC_ARG_WITH([openssl],AC_HELP_STRING([--with-openssl], [build with OpenSSL suppo
    [openssl="$withval"],[openssl="no"])
 
 if test "$openssl" = 'yes'; then
-  AC_CHECK_LIB([ssl], [SSL_library_init],,[AC_MSG_ERROR([ssl library missing])])
-  AC_CHECK_LIB([crypto], [CRYPTO_num_locks],,[AC_MSG_ERROR([crypto library missing])])
+  AC_CHECK_LIB([ssl], [SSL_CTX_new],,[AC_MSG_ERROR([ssl library missing])])
+  AC_CHECK_LIB([crypto], [CRYPTO_free],,[AC_MSG_ERROR([crypto library missing])])
 fi
 
 # GeoIP

